packages/components/src/notice/types.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode, SyntheticEvent } from 'react';
+
+type CommonNoticeActionProps = {
+	label: string;
+	className?: string;
+	noDefaultClasses?: boolean;
+	variant?: 'primary' | 'secondary' | 'link';
+};
+// `url` and `onClick` can both be provided, but `url` takes precedence. If
+// `url` is provided, the action's button will be rendered as an anchor and
+// `onClick` will be ignored.
+type NoticeActionWithURL = CommonNoticeActionProps & {
+	url: string;
+	onClick?: () => void;
+};
+type NoticeActionWithOnClick = CommonNoticeActionProps & {
+	url?: string;
+	onClick: ( event: SyntheticEvent ) => void;
```

Comment:
We should probably be a bit more specific, as this dictates to the consumer what would be available in the `event`.

```suggestion
	onClick: MouseEventHandler; // or even MouseEventHandler< HTMLButtonElement >
```

Reply:
I went with `MouseEventHandler`, because the 'button' is sometimes rendered as an anchor element - would that cause any problems with `MouseEventHandler<HTMLButtonElement>`? TypeScript didn't throw any errors, but it got me wondering 🤔

Reply:
It _could_, but I can't think of an obvious case off the top of my head and it should be fine in most cases. We can always update it if anybody has issues.

Reply:
Actually, I take that back - Some of `Snackbar`'s typing does take issue with `MouseEventHandler< HTMLButtonElement>`, I just didn't see it before. Specifically, Snackbar's `onActionClick` prop types the event as a `MouseEvent`,  which leaves it missing all of the button-specific props. We can resolve that typing `event` as `MouseEvent< HTMLButtonElement >`.

I've included that in da75964af2. Unless you'd prefer to hold off/revisit in a followup?

Reply:
I'm good for "enforcing" `HTMLButtonEvent` — a `button` seems like the obvious choice here, and (as also Lena said) I can't really think of an edge case where that would be a limitation.

Let's keep it as-is (i.e. with `HTMLButtonElement`), we can always revise this if it becomes problematic.